File: src/main/java/com/gildedrose/java/RegularItem.java
package com.gildedrose.java;

public class RegularItem extends Item {
    public RegularItem(String name, int sellIn, int quality) {
        super(name, sellIn, quality);
    }

    @Override
    protected void updateQuality() {
        decreaseQuality();
        if (sellIn < 0) {
            decreaseQuality();
        }
    }
}

File: src/main/kotlin/com/gildedrose/kotlin/RegularItem.kt
package com.gildedrose.kotlin

class RegularItem(name: String, sellIn: Int, quality: Int) : Item(name, sellIn, quality) {
    override fun updateQuality() {
        decreaseQuality()
        if (sellIn < 0) {
            decreaseQuality()
        }
    }
}